devicetypes/smartthings/zigbee-scene-keypad.src/zigbee-scene-keypad.groovy

```diff
@@ -0,0 +1,151 @@
+		}
+        
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat") {
+			state "default", action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+		main (["button"])
+		details(["button", "refresh"])
+	}
+}
+
+def parse(String description) {
+	def map = zigbee.getEvent(description)
+	def result = map ? map : parseAttrMessage(description)
+	if(result?.name == "switch") {
+		result = createEvent(descriptionText: "Wake up event came in", isStateChange: true)
+	}
+	log.debug "Description ${description} parsed to ${result}"
+	return result
+}
+
+def parseAttrMessage(description) {
+	def descMap = zigbee.parseDescriptionAsMap(description)
+    
+    	Map eventDescMap = zigbee.parseDescriptionAsMap(description)
+	log.debug "eventDescMap is $eventDescMap"
+    
+	def map = [:]
+	if (descMap?.clusterInt == 0x0017 || descMap?.clusterInt == 0xFE05) {
+	   def event = [:]
+       def buttonNumber
+       if(descMap?.clusterInt == 0x0017) {
+       		buttonNumber = Integer.valueOf(descMap.data[0]) 
+       } else if (descMap?.clusterInt == 0xFE05) {
+       		buttonNumber = Integer.valueOf(descMap?.value)
+       }
+       log.debug "Number is ${buttonNumber}"
+       event = createEvent(name: "button", value: "pushed", data: [buttonNumber: buttonNumber], descriptionText: "pushed", isStateChange: true)
+       if(buttonNumber != 1) {
+       		sendEventToChild(buttonNumber, event)
+       }else{
+       		sendEvent(event)
+       }
+    }
+}
+
+def getButtonEvent(descMap) {
+	if (descMap.commandInt == 1) {
+		getButtonResult("press")
+	}
+	else if (descMap.commandInt == 0) {
+		def button = buttonMap[device.getDataValue("model")][descMap.sourceEndpoint]
+		getButtonResult("release", button)
+	}
+}
+
+def sendEventToChild(buttonNumber, event) {
+	String childDni = "${device.deviceNetworkId}:$buttonNumber"
+	def child = childDevices.find { it.deviceNetworkId == childDni }
+	child?.sendEvent(event)
+}
+
+def refresh() {
+	return zigbee.enrollResponse()
+}
+
+def ping() {
+	refresh()
+}
+
+def configure() {
+	return configureReporting() + zigbee.enrollResponse()
+}
+
+def installed() {
+    def numberOfButtons = getChildCount()
+	sendEvent(name: "numberOfButtons", value: numberOfButtons, displayed: false)
+	sendEvent(name: "checkInterval", value: 2 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+	if(!childDevices) {
+		addChildButtons(numberOfButtons)
+	}
+	sendEvent(name: "supportedButtonValues", value: ["pushed"])
+}
+
+def updated() {
+	runIn(2, "initialize", [overwrite: true])
+}
+
+def initialize() {
+	
+}
+
+private addChildButtons(numberOfButtons) {
+	for(def i : 2..numberOfButtons) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$i"
+			def componentLabel = (device.displayName.endsWith(' 1') ? device.displayName[0..-2] : device.displayName) + "${i}"
+			addChildDevice("Child Button", "${device.deviceNetworkId}:${i}", device.hubId,
```

Comment:
I believe that `Child Button` supports held as well. You'll see "held" in the automation trigger list even though this device doesn't seem to generate "held" events.

Reply:
I think, according to my understanding from @dkirker that you'll need to send a "supportedButtonValues" event for each child.